src/main/java/de/photon/anticheataddition/modules/checks/inventory/InventoryAverageHeuristic.java

```diff
@@ -27,14 +32,11 @@ private InventoryAverageHeuristic()
         super("Inventory.parts.AverageHeuristic");
     }
 
-    private static boolean supportedClickType(ClickType type)
-    {
-        return switch (type) {
-            case DROP, RIGHT, LEFT, SHIFT_LEFT, SHIFT_RIGHT -> true;
-            default -> false;
-        };
+    private boolean supportedClickType(ClickType clickType) {
```

Comment:
Why did you replace a simple switch statement on a stable API enum that has implicit sorting and therefore highly optimized log(n) runtime with a ton of classes that make the code much less readable, causing a lot of overhead and force a linear search? I don't see any advantage of your method. 

I think you wanted to use a code pattern that is very good if you have a lot of classes and instead of using an instanceof check for every single one of them you use an abstract method. In that case, your idea is very good as it gets rid of all instanceofs in favor of small methods in each class. However, this is an enum and it is in Spigot, therefore unchangeable for me.